core_java/twitter/src/main/java/ca/jrvs/apps/twitter/dao/TwitterDao.java

```diff
@@ -0,0 +1,149 @@
+package ca.jrvs.apps.twitter.dao;
+
+import ca.jrvs.apps.twitter.dao.helper.HttpHelper;
+import ca.jrvs.apps.twitter.util.JsonParser;
+import ca.jrvs.apps.twitter.model.Tweet;
+import com.google.gdata.util.common.base.PercentEscaper;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.util.EntityUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+
+public class TwitterDao implements CrdDao<Tweet, String> {
+
+    //URI constants
+    private static final String API_BASE_URI = "https://api.twitter.com";
+    private static final String POST_PATH = "/1.1/statuses/update.json";
+    private static final String SHOW_PATH = "/1.1/statuses/show.json";
+    private static final String DELETE_PATH = "/1.1/statuses/destroy.json";
+
+    //URI symbols
+    private static final String QUERY_SYM = "?";
+    private static final String AMPERSAND = "&";
+    private static final String EQUAL = "=";
+
+    //Response code
+    private static final int HTTP_OK = 200;
+
+    private HttpHelper httpHelper;
+    private final PercentEscaper escaper = new PercentEscaper("", false);
+
+    @Autowired
+    public TwitterDao(HttpHelper httpHelper) {
+        this.httpHelper = httpHelper;
+    }
+
+    /**
+     * Create an Entity(Tweet) to the underlying storage
+     *
+     * @param tweet entity that need to be created
+     * @return entity created successfully
+     */
+    @Override
+    public Tweet create(Tweet tweet) {
+        //construct URI
+        URI uri;
+        try {
+            uri = getPostUri(tweet);
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException("Wrong input for tweet : " + e);
+        }
+
+        System.out.println(uri);
```

Comment:
use logger instead of print

Reply:
Just want to print it out and see if the URI is correct. I tried to print the URI by using logger, but it has to be String. Can I just delete it?

Reply:
yes. you can also consider using `logger.info` or `logger.debug` for this purpose